### PR TITLE
default branch defined for codecov.io

### PR DESCRIPTION
**Pull request by nupplaphil**
Follow-Up from #5570 
Solves #5568

### CODE REVIEW

**Comment by nupplaphil** [comment]
@tobiasd @rudloff @MrPetovan @annando 

Should I change the comment behavior?
https://docs.codecov.io/docs/pull-request-comments#section-disable-comment

what I would like:
- show just `diff` and `files` but **not** the graph
- just show it if something changes the coverage (so just css or html changes wouldn't get such a comment)

**Comment by tobiasd** [comment]
For me it is just pure SPAM. Just disabled my email notification from github.

**Comment by Rudloff** [comment]
Well, it is not really useful to have a code coverage tool if we don't display this information anywhere (and we are more likely going to introduce new untested code).

**Comment by tobiasd** [comment]
Travis is able to display the information without creating additional noise with every pull-request that is created. I got one mail per pull request before enabling this "service". Afterwards that amount doubled, the information is displayed as well in the web interface that I will most likely use to deal with the PR. This adds unneeded noise to the S/N ratio. If I want to have a look at the details of the Travis tests, I can follow the link to have a look at their page. Same would be true for a code coverage tool.

**Comment by nupplaphil** [comment]
We can try to use codecov - flags
https://docs.codecov.io/docs/flags

So codecov would create a check like travis-ci in the "check" section at the end of each PR, see:
https://files.readme.io/f89ab42-Screen_Shot_2016-08-17_at_3.54.13_PM.png

I'll try to customize the config

**Comment by Rudloff** [comment]
Having a check could indeed be a good way to display that info.

**Comment by nupplaphil** [comment]
- disabled the whole `comment` (=> report)
- Enabled `status/project` => checks the whole repo codecoverage
- Enabled `status/patch` => checks the code coverage of the current patch/pr
- I make two checks for `src/` and for the "rest" (`mod/` `mods/` `include/` )

I set 80% for `status/patch` so the check will fail if somebody commits *.php files without testings.
That doesn't mean that it's not mergable, but at least we can have an eye on it :-)

File: .codecov.yml [hunk 1]
```diff
@@ -1,2 +1,37 @@
...
+        flags: backend
+
+flags:
+  source:
+    paths:
+      - src/
+  backend:
+    paths:
+      - mod/
+      - mods/
```
Comment (tobiasd) [inline .codecov.yml]: Does this mean to include mods in the backend path or exclude it? mods is used for example files, don't think that needs to be covered.

Comment (nupplaphil) [reply .codecov.yml]: mod will be included
=> I'll remove it

Comment (tobiasd) [reply .codecov.yml]: `mod` is correct, but `mods` just holds examples.

**Pull request merged**